src/Microsoft.AspNetCore.Razor.Language/Intermediate/AddTagHelperHtmlAttributeIRNode.cs

```diff
@@ -9,8 +9,23 @@ namespace Microsoft.AspNetCore.Razor.Language.Intermediate
 {
     public sealed class AddTagHelperHtmlAttributeIRNode : RazorIRNode
     {
+        private List<RazorDiagnostic> _diagnostics;
+
         public override ItemCollection Annotations => ReadOnlyItemCollection.Empty;
 
+        public override IList<RazorDiagnostic> Diagnostics
+        {
+            get
+            {
+                if (_diagnostics == null)
+                {
+                    _diagnostics = new List<RazorDiagnostic>();
```

Comment:
If you want to, just make a copy/paste of the one from my PR and change the types. Let's keep it consistent